sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/codegen/GeneratedProjectionSuite.scala

```diff
@@ -83,6 +83,58 @@ class GeneratedProjectionSuite extends SparkFunSuite {
     assert(result === row2)
   }
 
+  test("SPARK-18016: generated projections on wider table requiring class-splitting") {
+    val N = 4000
+    val wideRow1 = new GenericInternalRow((1 to N).toArray[Any])
```

Comment:
I've cleaned up the test you comment on, and the one above it, since they both have the same structure, just different values for N:
https://github.com/apache/spark/pull/18075/commits/678b4ad770cbc891864f73d9b93c51b1ab79f6a5#diff-a14107cf4a4c41671bba24a82f6042d9R36